### ISSUE TITLE
asm is missing labels

### ISSUE THREAD

**Issue by lerno** [c3lang/c3c#888]
Labels need to be added. 

```
FOO:
   some_instruction;
```

**Comment by Book-reader** [comment]
It would also be useful to be able to get the address of labels:
```asm
uptr addr;
asm
{
  mov addr, &FOO;
  FOO:
}
```
declare data with something like this:
```asm
MY_NUMBER: .usz 0;

OTHER_NUMBERS:
  .usz 0;
  .usz 1;

mov MY_NUMBER, 123;
```
and use rip-relative addressing maybe with something similar to nasm's `rel`:
```asm
mov $rax, rel &LABEL;
```
the actual nasm assembler syntax would be closer to `mov [addr], FOO`, `mov [MY_NUMBER], 123`, and `mov $rax, [rel LABEL]` respectively, but this is closer to how it's already done with `mov my_var, some_value` so maybe it's better for the asm blocks?

**Comment by srkkov** [comment]
I agree with @Book-reader's suggestions, but I’d approach them a bit differently:

- rip-relative addressing static data should be the default for `X86_64`.

- A label is an address, I think it should just look like `mov [L2], $rax` and `mov $rax, L2`, where the former evaluates to a rip-rel offset, and the latter to an imm64. (Though in practice the latter would be better written as `lea $rax, [L2]`)

- Data declaring directives make more sense in an actual assembler, static data can already be placed in a custom section:
```
char[*] data @section("foo") = { 1, 2, 3, 4 };
``` 
and then we could do:
```
mov $al, [&data[2]];
```
But I still think we should have a byte-defining directive so the user can place instructions that may not be currently accessible. Something like:
```
db x"62 f1 74 d9 58 06"; // vaddps zmm0{k1}{z}, zmm1, [rsi]{1to16}
```
FASM-style assembly feels the most modern and does more work for you, so I think C3 should mirror it a bit.
